testing/mock/mock.go

```diff
@@ -194,7 +197,12 @@ func (am AppModule) OnChanCloseConfirm(sdk.Context, string, string) error {
 // OnRecvPacket implements the IBCModule interface.
 func (am AppModule) OnRecvPacket(ctx sdk.Context, packet channeltypes.Packet, relayer sdk.AccAddress) exported.Acknowledgement {
 	// set state by claiming capability to check if revert happens return
-	am.scopedKeeper.NewCapability(ctx, MockCanaryCapabilityName)
+	_, err := am.scopedKeeper.NewCapability(ctx, MockRecvCanaryCapabilityName+strconv.Itoa(int(packet.GetSequence())))
+	if err != nil {
+		// application callback called twice on same packet sequence
+		// must never occur
+		panic(err)
+	}
```

Comment:
Is this purely a way to check it isn't called twice?

Reply:
I'm fairly certain the capability name change needed to occur in order to receive 2 packets successfully on the same channel?

The panic was added to ensure the application callback isn't called twice as you say. Based on the structure of the code, it isn't even close to occurring, but if it did occurred it'd be a bad bug, so I figured it didn't hurt to add these panics